### PR TITLE
Tidy up release sections

### PR DESCRIPTION
**Pull request by Goooler**
Refs https://github.com/ComposeGears/Valkyrie/pull/121#discussion_r1713282566.

- Release snapshots instead of nightlys.
- Add more release template categories.

### CODE REVIEW

**Comment by Goooler** [comment]
https://github.com/ComposeGears/Valkyrie/actions/runs/10347838861/job/28638863729

**Comment by Goooler** [comment]
If we should remove `-p` for non-snapshot release? Which means it will not be a pre-release.

https://github.com/ComposeGears/Valkyrie/blob/0032ffc7dbcb0a9083fd3be8d710e2f62abb38e4/.github/workflows/release.yml#L77

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -34,12 +34,7 @@ jobs:
         run: |
           PLUGIN_NAME=$(grep "^PLUGIN_NAME=" gradle.properties | cut -d '=' -f 2)
           VERSION=$(grep "^VERSION_NAME=" gradle.properties | cut -d '=' -f 2)
-
-          if [[ $VERSION == *-SNAPSHOT ]]; then
-            RELEASE_NAME="Nightly"
-          else
-            RELEASE_NAME=$PLUGIN_NAME-$VERSION
-          fi
+          RELEASE_NAME=$PLUGIN_NAME-$VERSION
```
Comment (Goooler) [inline .github/workflows/release.yml:37]: And I believe we should remove `Valkyrie` prefix for tags, would be clear enough.

**Comment by egorikftp** [comment]
> If we should remove `-p` for non-snapshot release? Which means it will not be a pre-release.
> 
> https://github.com/ComposeGears/Valkyrie/blob/0032ffc7dbcb0a9083fd3be8d710e2f62abb38e4/.github/workflows/release.yml#L77

I think it should be a pre-release due to it's just a latest dev build to test if some feature works as expected, some issue fixed and so on. 
Release means stable build + published into marketplace

File: .github/workflows/release.yml (hunk 1 above)
Comment (egorikftp) [reply .github/workflows/release.yml:37]: Will the old tags have the old prefix? 
as for me prefix not critical)

**Review by egorikftp — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (Goooler) [reply .github/workflows/release.yml:37]: Yeah, old tags will keep their names, new tags will be like `0.6.0-SNAPSHOT`.

Comment (egorikftp) [reply .github/workflows/release.yml:37]: If it is really necessary, I don't mind changing it 🙂

**Comment by Goooler** [comment]
> I think it should be a pre-release due to it's just a latest dev build to test if some feature works as expected, some issue fixed and so on.
Release means stable build + published into marketplace

If I understand correctly, you made them pre-release for rearranging release notes, now this manual step has been replaced by this template.

- For a release, it will be sent to the marketplace and released from GH.
- For a snapshot, it will not be sent to the marketplace and pre-released from GH.

I mean we can remove `-p` for the second case here.

File: .github/workflows/release.yml (hunk 1 above)
Comment (Goooler) [reply .github/workflows/release.yml:37]: The version name for the tag is very clear here, we don't mix more than one project in this repository.

**Comment by egorikftp** [comment]
> I mean we can remove `-p` for the second case here.

Agree

File: .github/workflows/release.yml (hunk 1 above)
Comment (Goooler) [reply .github/workflows/release.yml]: 9a08ff5a169f0d7eff46446b8236b6eba2f37a4c

**Comment by Goooler** [comment]
https://github.com/ComposeGears/Valkyrie/actions/runs/10348719007/job/28641545831

File: .github/workflows/release.yml (hunk 1 above)
Comment (egorikftp) [reply .github/workflows/release.yml]: Seems like we need to manually delete the 'Nightly' release the first time, but I could be mistaken.

**Comment by egorikftp** [comment]
it compares old branch
image

**Comment by Goooler** [comment]
I see, deleted now.

**Comment by Goooler** [comment]
https://github.com/ComposeGears/Valkyrie/releases/tag/0.6.0-SNAPSHOT

**Comment by egorikftp** [comment]
Amazing)

**Pull request merged**